### PR TITLE
Fix demoting headers

### PR DESCRIPTION
**Pull request by okin**
### Pull Request Checklist

- [x] I’ve read the [guidelines for contributing](https://github.com/getnikola/nikola/blob/master/CONTRIBUTING.rst).
- [x] I updated AUTHORS.txt and CHANGES.txt (if the change is non-trivial) and documentation (if applicable).
- [x] I tested my changes.

### Description

The tests for header demotion weren't running and after activating them they failed.
Tests have been converted to make use of pytests parametrization feature to avoid duplication of test code.

The demotion function has been correct so that all cases work as expected.

### CODE REVIEW

File: nikola/utils.py [hunk 1]
```diff
@@ -1381,14 +1381,26 @@ def demote_headers(doc, level=1):
...
+            # html headers can't go lower than 1
+            after = 1
+        if after > 6:
+            # html headers go until 6
+            after = 6
+
+        if before == after:
+            continue
+
+        elements = doc.xpath('//h%i' % before)
```
Comment (Kwpolska) [inline nikola/utils.py]: We prefer modern string formatting, `doc.xpath('//h{}'.format(before))`

File: nikola/utils.py [hunk 2]
```diff
@@ -1381,14 +1381,26 @@ def demote_headers(doc, level=1):
...
+        for element in elements:
+            element.tag = 'h%i' % after
```
Comment (Kwpolska) [inline nikola/utils.py]: (Also here, `'h{}'.format(after)`)

**Comment by okin** [comment]
I'm totally fine with using the newer str.format. Code has been updated!

**Comment by Kwpolska** [comment]
Thanks for fixing this!

**Pull request merged**